prow/pod-utils/downwardapi/jobspec.go

```diff
@@ -164,6 +170,9 @@ func EnvForType(jobType prowapi.ProwJobType) []string {
 
 // getRevisionFromRef returns a ref or sha from a refs object
 func getRevisionFromRef(refs *prowapi.Refs) string {
+	if refs == nil {
```

Comment:
Nit: Before we could either panic or return a non-empty string. But now we can return either an empty string or a non-empty string. The current code is an improvement, so LGTM. BUT it would be better to make this return a (string, error) tuple instead instead of overloading the nil refs condition into the single `string` variable (unless we already do this elsewhere...).